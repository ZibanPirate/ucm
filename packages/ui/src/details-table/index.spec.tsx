import { render } from "@testing-library/react";

import { DetailsTable, DetailsTableRow } from ".";

describe(`Testing component '${DetailsTable.name}' :`, () => {
  it(`should render with provided linkWrapper as Fragment`, () => {
    const detailsRows: DetailsTableRow[] = [
      {
        label: "Model details",
        fields: {
          make: { label: "Make" },
          model: { label: "Model" },
        },
      },
      {
        label: "Finance",
        fields: {
          price: { label: "Price", mapper: (value) => `â‚¬${(value as number).toFixed(2)}` },
        },
      },
    ];
    const car = {
      make: "BMW",
      model: "M2",
      price: 39000,
    };

    const { container } = render(<DetailsTable rows={detailsRows} values={car} />);
    expect(container).toMatchSnapshot();
  });
});
