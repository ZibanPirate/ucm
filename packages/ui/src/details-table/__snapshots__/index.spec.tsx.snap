// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component 'DetailsTable' : should render with provided linkWrapper as Fragment 1`] = `
<div>
  <div>
    <div>
      <span
        style="font-size: medium; margin: .3rem; display: flex;"
      >
        Model details
      </span>
      <hr />
      <div
        style="display: flex; flex-direction: row; flex: 1; margin: 0px .3rem; justify-content: space-between;"
      >
        <span
          style="font-size: small; margin: initial; display: flex;"
        >
          Make
        </span>
        <span
          style="font-size: small; margin: initial; display: flex;"
        >
          BMW
        </span>
      </div>
      <br />
      <div
        style="display: flex; flex-direction: row; flex: 1; margin: 0px .3rem; justify-content: space-between;"
      >
        <span
          style="font-size: small; margin: initial; display: flex;"
        >
          Model
        </span>
        <span
          style="font-size: small; margin: initial; display: flex;"
        >
          M2
        </span>
      </div>
      <br />
    </div>
    <div>
      <span
        style="font-size: medium; margin: .3rem; display: flex;"
      >
        Finance
      </span>
      <hr />
      <div
        style="display: flex; flex-direction: row; flex: 1; margin: 0px .3rem; justify-content: space-between;"
      >
        <span
          style="font-size: small; margin: initial; display: flex;"
        >
          Price
        </span>
        <span
          style="font-size: small; margin: initial; display: flex;"
        >
          â‚¬39000.00
        </span>
      </div>
      <br />
    </div>
  </div>
</div>
`;
